hypothesis-python/tests/py3/test_lookup.py

```diff
@@ -563,3 +563,16 @@ def bar(self):
 @given(st.from_type(AbstractBar))
 def test_cannot_resolve_abstract_class_with_no_concrete_subclass(instance):
     assert False, "test body unreachable as strategy cannot resolve"
+
+
+@pytest.mark.parametrize(
+    "typ", [typing.Hashable, typing.Sized], ids=["Hashable", "Sized"]
+)
+@given(data=st.data())
+def test_inference_on_generic_collections_abc_aliases(typ, data):
+    # regression test for inference bug on types that are just aliases
+    # types for simple interfaces in collections abc and take no args
+    # the typing module such as Hashable and Sized
+    # see https://github.com/HypothesisWorks/hypothesis/issues/2272
+    value = data.draw(st.from_type(typ))
```

Comment:
cool cheers i was unsure if i should include the higher level inference thing or just call from_type and using data strategy is much neater.

Reply:
`infer` just means "use `from_type` behind the scenes" :smile: 

And putting `@given` and `@pytest.mark.parametrize` on the same function means that the Hypothesis example database gets separate entries for each param, rather than having them all clobber each other.

Reply:
aw awesome good to know.. tbh  i didn't know they could be combined and i use this hypothesis all the time.